subject.py

```diff
@@ -0,0 +1,178 @@
+
+
+
+@schema
+class Species(dj.Manual):
+    # <class 'subjects.models.Species'>
+    definition = """
+    species_id:			int             # species id
+    ---
+    binomial:			varchar(255)	# binomial
+    display_name:		varchar(255)	# display name
+    """
+
+
+@schema
+class Strain(dj.Manual):
+    # <class 'subjects.models.Strain'>
+    definition = """
+    strain_id:			int             # strain id
+    ---
+    descriptive_name:		varchar(255)	# descriptive name
+    description:    		varchar(255)	# description
+    """
+
+
+@schema
+class Sequence(dj.Manual):
+    # <class 'subjects.models.Sequence'>
+    definition = """
+    sequence_id:		int             # sequence id
+    ---
+    base_pairs:			varchar(255)	# base pairs
+    description:		varchar(255)	# description
+    informal_name:		varchar(255)	# informal name
+    """
+
+
+@schema
+class Allele(dj.Manual):
+    # <class 'subjects.models.Allele'>
+    definition = """
+    allele_id:			int             # allele id
+    ---
+    standard_name:		varchar(255)	# standard name
+    informal_name:		varchar(255)	# informal name
+    """
+
+
+@schema
+class Line(dj.Manual):
+    # <class 'subjects.models.Line'>
+    # TODO: autoname index: basically autogenerates a unique sequence
+    definition = """
+    -> Species
+    -> Strain
+    -> Sequence
+    line_id:				int             # line id
+    ---
+    name:				varchar(255)	# name
+    description:			varchar(255)	# description
+    target_phenotype:			varchar(255)	# target phenotype
+    auto_name:				varchar(255)	# auto name
+    is_active:				boolean		# is active
+    subject_autoname_index:		integer		# subject autoname index
+    breeding_pair_autoname_index:	integer		# breeding pair autoname index
+    litter_autoname_index:		integer		# litter autoname index
+    """
+
+
+@schema
+class Source(dj.Manual):
+    # <class 'subjects.models.Source'>
+    definition = """
+    source_id:				int             # source id
+    ---
+    name:				varchar(255)	# name
+    description:			varchar(255)	# description
+    """
+
+
+@schema
+class SubjectRequest(dj.Manual):
+    # <class 'subjects.models.SubjectRequest'>
+    definition = """
+    -> reference.User
+    -> Line
+    subject_request_id:			int     	# subject request id
+    ---
+    count:				integer		# count
+    date_time:				date		# date time
+    due_date:				date		# due date
+    description:			varchar(255)	# description
+    """
+
+
+@schema
+class Subject(dj.Manual):
+    # <class 'subjects.models.Subject'>
+    # todo:
+    # - where did responsible_user go? RETEST ADDED
+    # - -> SubjectRequest or track subjects as part table of subject request?
+    #      RETEST ADDED
+
+    definition = """
+    subject_id:			int                     # subject id
+    ---
+    nickname:			varchar(255)		# nickname
+    -> Line
+    sex:			enum("M", "F", "U")	# sex
+    birth_date:			date			# birth date
+    death_date:			date			# death date
```

Comment:
Inclusion of field that has to be added later is typically a bad idea as this implicitly requires use of `UPDATE` on the table. Let's separate `death_date` into separate `Manual` table.

Reply:
should birth/death be separate parts of Subject?

Reply:
`birth_date` can stay as part of the subject but death date will be better as a separate table (not not a `Part` table). This is because you would usually not know the death date of the subject at the time of the subject entry whereas birthday is usually known (or if not known can remain empty - depending on how strict the designer wants this field to be filled).